File: Java言語/基礎知識用の勉強/2配列勉強/配列.java
class 配列 {
	public static void main(String[] args){
		System.out.print("hello World.\n");
		int asakuno = 20210918;

		int[] tomohiro = {
					20210918,
					20210919,
					20210920,
					};
		int[] hoge;	// 宣言と代入を別にできるが、別にする理由はあるのか？
		hoge = new int[]{
					13310679,
					54755031,
					64546000,
					24330270,
					tomohiro[1]
					};
		var asakunoarray = new int[] {
					54755032,
					64546030,
					24330271,
					};	// 配列の場合、型推論型を使う利点は潰されているな。

		System.out.println( tomohiro[0] );	// 20210918
		System.out.println( hoge[3] );		// 24330270
		// System.out.println( hoge[5] );	// エラー		要素外への干渉は不可。
					// Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 5 out of bounds for length 5
					// 	at 配列.main(配列.java:22)
		System.out.println( hoge[4] );		// 20210919
		System.out.println( asakunoarray[1] );	// 64546030
		System.out.println( asakunoarray.length );	// 3	←☆またもや,ややこしい(当たり前の結果だが)。
		//asakunoarray[3] = 20210918;	// エラー		配列を拡張することは出来ない。
					// Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 3 out of bounds for length 3
					// 	at 配列.main(配列.java:33)

	}
}

// vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=java noexpandtab:

File: Go言語/基礎知識用の勉強/2配列勉強/実行済み/配列.go
package main

import "fmt"

func main() {
	// 以下、配列宣言後に代入。
	var asakuno [10]int;
	asakuno[0] = 20210928;
	fmt.Println(asakuno[0]);	// 20210928
	fmt.Println(asakuno[1]);	// 0
	// fmt.Println(asakuno[10]);	// # command-line-arguments	invalid array index 10 (out of bounds for 10-element array)
	fmt.Println("-----------------------");

	// 以下、配列宣言と同時に代入(初期化)。
	var tomohiro = [10]int {20210928, 20210929 };
	fmt.Println(tomohiro[0]);	// 20210928
	fmt.Println(tomohiro[1]);	// 20210929
	fmt.Println(tomohiro[9]);	// 0
	// fmt.Println(tomohiro[10]);	// 上記と同様のエラー。

	// 以下、初期化方法失敗。
	// var tomohiro2[10] = int {20210928, 20210929 };	// syntax error: unexpected =, expecting type
	// var tomohiro2[10] int = {20210928, 20210929 };	// syntax error: unexpected {, expecting expression
	// var tomohiro2 [10]int {20210928, 20210929 };		// syntax error: unexpected { at end of statement

	// 以下、別の方法での初期化。
	asakunotomohiro := [10]string {"朝来野", "智博", };
	fmt.Println(asakunotomohiro);		// [朝来野 智博        ]
	fmt.Println(asakunotomohiro[0]);	// 朝来野
	fmt.Println(asakunotomohiro[1]);	// 智博
	fmt.Println(asakunotomohiro[2]);	// 空文字列

	// 以下、要素数をコンパイラ任せにする。
	tomohiroasakuno := [...]string {"朝来野", "智博", };
	fmt.Println(tomohiroasakuno);		// [朝来野 智博]
	fmt.Println(tomohiroasakuno[0]);	// 朝来野
	fmt.Println(tomohiroasakuno[1]);	// 智博
	// fmt.Println(tomohiroasakuno[2]);	// invalid array index 2 (out of bounds for 2-element array)

	// 以下、配列ではなくスライス扱いになる。
	sliceasakuno := []string {"スライス", "asakuno", };
	fmt.Println(sliceasakuno);		// [スライス asakuno]
	fmt.Println(sliceasakuno[0]);	// スライス
	fmt.Println(sliceasakuno[1]);	// asakuno
	// fmt.Println(sliceasakuno[2]);	// panic: runtime error: index out of range [2] with length 2
}

/* vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=go noexpandtab: */
